src/main/java/gyro/google/compute/SnapshotSchedulePolicySnapshotProperties.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2020, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.compute;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.google.api.services.compute.model.ResourcePolicySnapshotSchedulePolicySnapshotProperties;
+import gyro.core.resource.Diffable;
+import gyro.core.resource.DiffableField;
+import gyro.core.resource.DiffableType;
+import gyro.google.Copyable;
+
+public class SnapshotSchedulePolicySnapshotProperties extends Diffable implements Copyable<ResourcePolicySnapshotSchedulePolicySnapshotProperties> {
+
+    private Boolean guestFlush;
+    private Map<String, String> labels;
+    private List<String> storageLocations;
+
+    /**
+     * Perform a guest aware snapshot.
+     */
+    public Boolean getGuestFlush() {
+        return guestFlush;
+    }
+
+    public void setGuestFlush(Boolean guestFlush) {
+        this.guestFlush = guestFlush;
+    }
+
+    /**
+     * Labels to apply to scheduled snapshots.
+     */
+    public Map<String, String> getLabels() {
+        return labels;
+    }
+
+    public void setLabels(Map<String, String> labels) {
+        this.labels = labels;
+    }
+
+    /**
+     * Cloud Storage bucket storage location of the auto snapshot (regional or multi-regional).
+     */
+    public List<String> getStorageLocations() {
+        return storageLocations;
+    }
+
+    public void setStorageLocations(List<String> storageLocations) {
+        this.storageLocations = storageLocations;
+    }
+
+    @Override
+    public String primaryKey() {
+        List<String> key = new ArrayList<>();
+        DiffableType type = DiffableType.getInstance(this);
+        List<DiffableField> fields = type.getFields();
+        for (DiffableField f : fields) {
+            Object value = f.getValue(this);
+            if (value != null) {
+                key.add(f.getName() + "=" + f.getValue(this));
+            }
+        }
+
+        if (key.size() > 0) {
+            return key.stream().collect(Collectors.joining(", "));
+        }
+        return "";
```

Comment:
Even though this does work, it will be better to refactor this to add the fields individually.

Alternatively as this complex type is used as a single field and not a list, just returning `""` will suffice as well.

Reply:
Done.